spec/lib/payrix/resource/base_spec.rb

```diff
@@ -0,0 +1,20 @@
+RSpec.describe Payrix::Resource::Base do
+  Payrix::Resource::Tests = Class.new(Payrix::Resource::Base) do
+    def initialize(params)
+      super(params, [:status, :amount])
+
+      @resource_name = 'test'
+    end
+  end
+
+  let(:resource) { Payrix::Resource::Tests.new(status: '1', amount: '500') }
+
+  it 'raises Payrix::Exceptions::ApiError with the message which includes the errors payload' do
+    expect { resource.create }
+      .to raise_error(Payrix::Exceptions::ApiError)
+      .with_message(
+        'There are errors in the response, '\
+        '[{"code"=>13, "severity"=>4, "msg"=>"No such action and/or resource", "errorCode"=>"invalid_resource"}]',
+      )
```

Comment:
Is this test executing against a real Payrix API? Executing real network requests in the test suite may pose security vulnerabilities for the machine running the tests, and may also cause flakiness if the API were to ever change.

I suggest we include a gem like `webmock` as a testing dependency, or find a way to stub the base response with a dummy response.

Let me know what you think.

Reply:
Good catch, that was an oversight on my end. Fixed.